docs/decisions/0035-embedding-abstraction.md

```diff
@@ -0,0 +1,257 @@
+---
+# These are optional elements. Feel free to remove any of them.
+status: proposed
+contact: Krzysztof318
+date: 2024-03-02
+deciders: ???
+consulted: ???
+informed: ???
+---
+
+# Embeddings Generation in Semantic Kernel
+
+## Context and Problem Statement
+
+### General Information
+
+The current abstraction of generating embeddings only allows for generation from text data.\
```

Comment:
That's a case for `ITextEmbeddingGenerationService` but `IEmbeddingGenerationService<TValue, TEmbedding>` is not coupled to particular data type. 

Today, if you create new connector that implements `IEmbeddingGenerationService<ImageContent, float>`, it should work as expected. But this interface doesn't support metadata in response and does not accept execution settings, but these are separate problems. So, maybe it will be good to re-phrase `Content and Problem Statement` section to avoid confusion.